chalk-solve/src/clauses/builtin_traits/fn_family.rs

```diff
@@ -86,7 +85,7 @@ pub fn add_fn_trait_program_clauses<I: Interner>(
     builder: &mut ClauseBuilder<'_, I>,
     well_known: WellKnownTrait,
     self_ty: Ty<I>,
-) -> Result<(), Floundered> {
+) {
```

Comment:
Refactor: made `add_fn_trait_program_clauses` return `()` because it's infallible.